### ISSUE TITLE
support string as primary key

### ISSUE THREAD

**Issue by newlix** [tekartik/sqflite#60]
Can I use string id instead of integer? I am new to the dart language. I expect the rawInsert should return Future<T> instead of Future<int>.  So the type system can infer the id type with the return var type.
```dart
int id = txt.rawInsert(...)
// or
String id = txt.rawInsert(...)
```

**Comment by alextekartik** [comment]
Thanks for your report. Actually the id returned is what SQLite called ROWID which is typically mapped to your primary key if it is an INTEGER (or created if your key is not an INTEGER). There is a workaround in SQLite starting 3.8.2 to create a table WITHOUT ROWID, in this case (in my experiments) it will always returns 1. 

For better information see the notes here: https://www.sqlite.org/withoutrowid.html

I also created some tests on this here:
https://github.com/tekartik/sqflite/blob/510f722d3ad166d44a3f42965333989cfe4bdc49/example/lib/raw_test_page.dart#L548-L594

Basically what I would suggest if you use a TEXT as a primary key is to simply ignored the returned value (since you know the primary key anyway)

**Comment by newlix** [comment]
Thanks for the thorough explanation.
